slang-gfx.h

```diff
@@ -1912,6 +1910,23 @@ class ITransientResourceHeap : public ISlangUnknown
         0xcd48bd29, 0xee72, 0x41b8, { 0xbc, 0xff, 0xa, 0x2b, 0x3a, 0xaa, 0x6d, 0xeb } \
     }
 
+class ID3D12TransientResourceHeap : public ISlangUnknown
+{
+public:
+    enum class DescriptorType
+    {
+        ResourceView, Sampler
+    };
+    virtual SLANG_NO_THROW Result SLANG_MCALL allocateTransientDescriptorTable(
+        DescriptorType type,
+        uint32_t count,
+        uint64_t& outDescriptorOffset,
+        void** outD3DDescriptorHeapHandle) = 0;
```

Comment:
I think this is completely reasonable to have as a low-level interface, but I wonder if this is a case where we ultimately need to have a "mid-level" `gfx` interface that represents a D3D12/Vulkan descriptor table/set, and allow allocation of those from a transient heap in an API-independent manner (ignoring the way that can't be ported to the other APIs easily).

*If* we went down the road of such a mid-level API it would not make sense to have this `DescriptorType` enumeration, which requires every table to either be a resource-view table or a sampler table, given that Vulkan supports sets that mix these cases. In the portable mid-level version, a D3D12 "descriptor set" would consist of zero or one resource-view table and zero or one sampler table.

I guess the problem is that going down that road quickly requires us to have both an abstraction for descriptor sets/tables *and* one for the layouts for those sets/tables (where VK would require the layout, but D3D largely wouldn't care). A user who knows they are on D3D12 could then end up needing/wanting this lower-level API anyway... sigh.